### ISSUE TITLE
Create Feature Index page

### ISSUE THREAD

**Issue by KianBadie** [KianBadie/cell-atlas#11]
Create Feature Index appendix page. The layout will be a single page list similar to the other appendix pages. Will determine specifics of layout through discussion.

Screenshot from bookdown site:
![Screenshot from 2020-12-03 10-28-09](https://user-images.githubusercontent.com/18221058/101072058-4966db00-3552-11eb-88f4-b9a8d7760049.png)

**Comment by KianBadie** [comment]
@coiko For this appendix page, I'm assuming it is going to be another single page list. Would you like the layout of this page to be 
more similar to the profiles page (with expanding dropdowns for a feature like "Archaella") or the references page (where everything is visible in the list)? Or were you thinking of something different?

**Comment by coiko** [comment]
@KianBadie Yes, exactly.  It would be great if it was the same style as the Profiles page, with the expanding sections.

And yes, only the "More" flavor of Learn More button has features referenced in the index, not the schematics.

As for the figure numbers, that's a convention from bookdown that I don't think we need to continue unless it would be useful to you.  I always thought the figure numbers were confusing because they didn't match the section numbers, so we tried to suppress them and only used them for the linking feature.  But if you can link directly to the section/subsection, that would be much better.

And for the list, it would be awesome to show the section and (when applicable) subsection.  But if it's easier to just show the species name and link from that (without any numbering or section titles), that would be perfectly fine for the first version.  Thanks!

**Comment by KianBadie** [comment]
@coiko I added a first go at the feature index page. I also added some functionality to auto-open the subsection if the linked feature was a subsection (shown in the gif). What do you think of this page? Any comments or critiques?

I think there might be some features that have non-working links to figures. For example, it seems like the Chlorosomes feature is trying to link to a figure "4-4", but it seems that section 4-4 does not exists in the source txt [file](https://github.com/KianBadie/cell-atlas/blob/main/04_Growth.txt). The [page](https://cellstructureatlas.org/book/chlorosomes.html) on the bookdown site for the Chlorosomes feature also shows how it's having trouble connecting the reference to the correct figure. I'm guessing section 4-4 was removed, but the figure number wasn't updated. If this is the case, maybe at some point in the future we could create some sort of system where the feature list would be conveniently and dynamically created rather than dependent on a manual solution. I have a list of feature index entries that are similar to what I just described, so I can remove those references or share the list if interested (it's a pretty short list, so no worries).  

## Gif
![Peek 2020-12-03 15-38](https://user-images.githubusercontent.com/18221058/101103765-7df08c00-357e-11eb-94b7-e0de41f7b9c1.gif)
## Screenshot
![Screenshot from 2020-12-03 15-57-52](https://user-images.githubusercontent.com/18221058/101104602-54386480-3580-11eb-9a79-7ca4593d2328.png)

**Comment by coiko** [comment]
@KianBadie This is beautiful!  It's absolutely perfect!  And I love that it auto-opens the relevant Learn More subsections.

Sorry about the missing links.  Yes, we pulled out two pages (4-4 and a More subsection of 5-6) after we shared the book draft with the contributors.  (It's a long story, but one of the contributors asked us to hold off on showing something because they were publishing a research article about it and didn't want a competing group to see it here first.  As soon as the article is published, we'll put the pages back in.)  If there are other entries missing, yes, please let me know.  For now, we can just remove the entries, but yes, I agree it would be great to come up with a better system in the future.  With your neat structure of metadata for each section, maybe we could add a field with the feature keywords that could be dynamically compiled into the index?  I have a spreadsheet tracking all the features that appear in each movie, so it would be easy for me to transfer the information.  Anyway, we can discuss that later.  For now, this is fantastic.  Thank you!

**Comment by KianBadie** [comment]
@coiko Thank you! I'll leave it in this style for now then!

And I see now, that makes sense. The other entries that seem to be missing are: 
- Methanobacterium formicicum (fig:2-2c)
- Chloroflexi (fig:4-4)
- Chloroflexi (fig:5-6b)

The second 2 make sense as they line up with your explanation. Does the 1 one make sense? Based on the [txt file](https://github.com/KianBadie/cell-atlas/blob/main/02_Cells.txt) for chapter 2, it looks like it just goes up to 2b. 

And sounds good! That future systems sounds like a fun project. Yeah, I was thinking of something like you described with the metadata. Oh yeah that spreadsheet would be a great tool to use in implementing that! And no problem!

**Comment by KianBadie** [comment]
@coiko I think some links in the feature index stopped working since we reordered chapter 4. We could fix it manually, but I could also regenerate the feature index list, depending on what changes you made to the list. I remember yesterday you added changes to it. Was it just changes in title or capitalization or something more?

**Comment by coiko** [comment]
@KianBadie Ah, sorry about that!  I duplicated all the changes I made in the 11_Features.Rmd file, so if you have the latest version (from yesterday), it should be fine to regenerate it.  And it would be awesome to change the casing issue (I didn't fix the capitalization)!

The changes I made were:

- took out Chlorosomes
- changed "Unknown structures" to "Unidentified structures"
- swapped that section with "Unipolar polysaccharide" for alphabetization
- removed Shewanella oneidensis - 10.1 Phage from Type II secretion systems
- added Hylemonella gracilis - 3.3 Length to Nucleoid
- added Hylemonella gracilis - 6.3 Flagellar Assembly to Nucleoid
